### ISSUE TITLE
Program Breaking issue with 1.55 Equipment Panel

### ISSUE THREAD

**Issue by HammerGS** [MegaMek/megameklab#16]
Downloaded a fresh version of 1.55 and started it.  When you try to switch to the equipment panel the program freezes for a second and won't go to that panel.

See attached log for errors.

[megameklablog.txt](https://github.com/MegaMek/megameklab/files/310551/megameklablog.txt)

**Comment by pheonixstorm** [comment]
I looked into this a bit and found no reason for the behavior. I did find that if you try to load a file from cache rather than building a new unit the cache does not get created and the program hangs with the first error in your posted log. There wasn't a lot of code change between 0.1.54 and 0.1.55 coming from MegaMek

**Comment by BlivetWidget** [comment]
I get the error on a fresh unpacking of 0.1.55 and not with 0.1.54.

**Comment by pheonixstorm** [comment]
Yeah 0.1.54 worked, but something changed between then and the new release. Haven't found what so I hope Arlith can get it figured out before the next release.

**Comment by Dylan-M** [comment]
https://git-scm.com/docs/git-bisect is your friend.

**Comment by pheonixstorm** [comment]
Found it, posted a bug report in MM tracker MegaMek/megamek#254

**Comment by arlith** [comment]
The problem is that MM has started using libxstream for other serialization, which is now used in MML, but the MML project doesn't have the libxstream library.  I've got this fixed locally, but I'm waiting to push the fix.  In the meantime, as a work around, you can copy the libxstream.jar file from MM's lib directory into MML's lib directory and it should work.

**Comment by pheonixstorm** [comment]
It doesn't, I tried that first to be sure. Thats why I posted this in the MM tracker

**Comment by arlith** [comment]
The issue is most assuredly a missing libxstream.jar, you said it yourself.  Here's the stack trace:

```
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: com/thoughtworks/xstream/converters/Converter
	at megamek.common.EquipmentType.getGameOptions(EquipmentType.java:1005)
	at megamek.common.weapons.ISERLargeLaser.getLongRange(ISERLargeLaser.java:67)
	at megameklab.com.util.EquipmentTableModel.getValueAt(EquipmentTableModel.java:314)
	at javax.swing.JTable.getValueAt(Unknown Source)
```

What does this mean?  It's saying that when `EquipmentType.java` line 1005 is reached, it requires a `com.thoughtworks.xstream.converts.Converter` class.  What's happening at `EquipmentType.java` line 1005?

`if (Server.getServerInstance() == null) {`

Ok, so when did this problem occur? Megamek/megamek@2a466381c3166438c3e10102a9a0d40734a4d001, according to you.  What happened in that commit? A reference to  `com.thoughtworks.xstream.converts.Converter` was added in `megamek.server.Server`.  This issue is exactly a missing dependency, which is resolved by libxstream.jar.

There was an additional stack trace in the logfile that I fixed.  It was related to a missing UnitVerifierOptions.xml.  I copied the default version from MM's data/mechfiles directory into MML.

**Comment by pheonixstorm** [comment]
Unless there is a missing step to this I am still having the same problem on windows even after updating MML and compiling from the current HEAD. Same error Hammer posted above.
[megameklablog.txt](https://github.com/MegaMek/megameklab/files/317535/megameklablog.txt)

**Comment by arlith** [comment]
I pulled a new version of the MML repository on windows, built a jar file and ran it and it worked.  If you're getting a `java.lang.NoClassDefFoundError: com/thoughtworks/xstream/converters/Converter` error then it's not finding libxstream.  Try executing the MML jar from outside of Eclipse and see if you get the error.

You should have to add the jar file in your Eclipse project.  I don't really know how you have your projects setup (I only have one project: megamek.  Then, I linked to external source within that project.  So the build environment is all the same, which is why I sometimes miss bugs like this for MML, although that wasn't the reason in this case).

**Comment by pheonixstorm** [comment]
I compiled MML and unzipped it in another folder outside eclipse. I used the packaging_local.xml to build MML which did include xstrem in the lib folder. Did not try running it in eclipse.

Should the MML classpath also have xstream listed? That wasn't part of your fix but I can try adding it and see what result I get. Will also try updating everything on my old computer and see if it works there or not.

**Comment by pheonixstorm** [comment]
Update: If I run it within eclipse it works fine. If I compile a new zip and run it outside eclipse it fails with the original error.

**Comment by arlith** [comment]
Alright, looks like the Ant scripts don't generate the jar manifest automatically, but instead use `manifest.txt` in the root of the repo.  That needed to be updated.  The reason it runs from the root of the repo without packaging is because it uses the MegaMek.jar, which has lib/xstream-1.4.7.jar in its manifest.  I updated the classpath in MML's manifest to also include the xstream jar.

Additionally, I found that packaging_local.xml was broken.  It was still configured to generate the manifest automatically and didn't have any of the MML dependencies (also, the MM jar gets packaged into the MML jar, and its manifest is lost).  There was a commit a while back from Dylan that removed this automatic manifest generation in packaging.xml, but packaging_local.xml seems to have been missed.
